tests/splines/extrapolation_rule.cpp

```diff
@@ -336,10 +336,8 @@ static void ExtrapolationRuleSplineTest()
                 // Set coords_eval outside of the domain
                 ddc::get<I1>(coords_eval(e))
                         = xN<I1>() + (ddc::select<I1>(ddc::coordinate(e)) - x0<I1>()) + 1;
-#if defined(BC_GREVILLE)
                 ddc::get<I2>(coords_eval(e))
-                        = xN<I2>() + (ddc::select<I2>(ddc::coordinate(e)) - x0<I2>()) + 1;
-#endif
+                        = xN<I2>() + (ddc::select<I2>(ddc::coordinate(e)) - x0<I2>());
```

Comment:
```suggestion
                        = xN<I2>() + (ddc::select<I2>(ddc::coordinate(e)) - x0<I2>()) + 1;
```

The `+1` is to ensure we are strictly outside of the domain even for `e` on "left border" (`ddc::select<I2>(ddc::coordinate(e)) = x0<I2>()`)

Reply:
Ah. I removed it so that the solution in the periodic case would be found in `vals`. I see 3 possible solutions:
1. exclude the left border
2. exclude the right border and add `ddc::coordinate(e+1)-ddc::coordinate(e)`
3. Add 2*`xN<I2>()`
4. Use `evaluator` to calculate the solution on the fly instead of using `vals`

Which do you think would be best? I think 3. would be the simplest

Reply:
I agree 3. looks good

Reply:
Fixed [afb31fa](https://github.com/CExA-project/ddc/pull/364/commits/afb31fa412e18d9bc569b1c760374817c911745b)